test/simulator/timestamp/simulator_interface/simulator_interface.cpp

```diff
@@ -0,0 +1,185 @@
+        std::cout << "\n" << cli_str << " ";
+        std::getline(std::cin, text_line);
+
+        /* Extract number from the text. */
+        std::istringstream text_stream(text_line);
+        text_stream >> choice;
+
+        /* Validate the number of choice. */
+        if (choice < min || choice > max)
+            print_border_msg(
+              "Choose a number between " + std::to_string(min) + " and " + std::to_string(max),
+              RED);
+
+    } while (choice < min || choice > max);
+
+    return (choice);
+}
+
+void
+interface_session_management()
+{
+}
+
+void
+interface_set_timestamp()
+{
+}
+
+void
+interface_conn_query_timestamp()
+{
+}
+
+void
+interface_begin_transaction()
+{
+}
+
+void
+interface_commit_transaction()
+{
+}
+
+void
+interface_prepare_transaction()
+{
+}
+
+void
+interface_rollback_transaction()
+{
+}
+
+void
+interface_timestamp_transaction()
+{
+}
+
+void
+interface_session_query_timestamp()
+{
+}
+
+void
+print_rules()
+{
+}
+
+int
+main(int argc, char *argv[])
+{
+    bool exit = false;
+
+    do {
+        std::cout << std::endl;
+
+        std::cout << "1: Session Management " << std::endl;
+        std::cout << "2: [Conn] set_timestamp()" << std::endl;
+        std::cout << "3: [Conn] query_timestamp()" << std::endl;
+        std::cout << "4: [Session] begin_transaction()" << std::endl;
+        std::cout << "5: [Session] commit_transaction()" << std::endl;
+        std::cout << "6: [Session] prepare_transaction()" << std::endl;
+        std::cout << "7: [Session] rollback_transaction()" << std::endl;
+        std::cout << "8: [Session] timestamp_transaction()" << std::endl;
+        std::cout << "9: [Session] query_timestamp()" << std::endl;
+        std::cout << "10: print rules for timestamps" << std::endl;
+        std::cout << "11: exit" << std::endl;
+
+        int choice = choose_num(1, 11, "timestamp_simulator >>");
+
+        try {
+            switch (choice) {
+            case 1:
+                interface_session_management();
+                break;
+            case 2:
+                interface_set_timestamp();
+                break;
+            case 3:
+                interface_conn_query_timestamp();
+                break;
+            case 4:
+                interface_begin_transaction();
+                break;
+            case 5:
+                interface_commit_transaction();
+                break;
+            case 6:
+                interface_prepare_transaction();
+                break;
+            case 7:
+                interface_rollback_transaction();
+                break;
+            case 8:
+                interface_timestamp_transaction();
+                break;
+            case 9:
+                interface_session_query_timestamp();
+                break;
+            case 10:
+                print_rules();
+                break;
+            case 11:
+                exit = true;
+            }
+        } catch (std::string &exception_str) {
```

Comment:
Can the string be `const`?

Reply:
Updated in [d3b07ca](https://github.com/wiredtiger/wiredtiger/pull/8417/commits/d3b07cad2974c45eb19a02631df03808cff9758d)